common/scala/src/main/scala/whisk/core/database/CouchDbStoreProvider.scala

```diff
@@ -36,21 +47,18 @@ object CouchDbStoreProvider extends ArtifactStoreProvider {
     actorSystem: ActorSystem,
     logging: Logging,
     materializer: ActorMaterializer): ArtifactStore[D] = {
-    require(config != null && config.isValid, "config is undefined or not valid")
+    val dbConfig = loadConfigOrThrow[CouchDbConfig](ConfigKeys.couchdb)
     require(
-      config.dbProvider == "Cloudant" || config.dbProvider == "CouchDB",
-      "Unsupported db.provider: " + config.dbProvider)
-    assume(
-      Set(config.dbProtocol, config.dbHost, config.dbPort, config.dbUsername, config.dbPassword, name(config))
-        .forall(_.nonEmpty),
-      "At least one expected property is missing")
+      dbConfig.provider == "Cloudant" || dbConfig.provider == "CouchDB",
+      s"Unsupported db.provider: ${dbConfig.provider}")
```

Comment:
Is there anything left that depends on `provider`? If not, please rip it out.

Reply:
Not sure what is the intention of provider so far. In code this param is not used. I just kept existing logic as is with switch to pureconfig. If provider is not required (if others can confirm) I can then remove that also

Reply:
@rabbah can you confirm? I think we used to have a distinction but killed all of that long ago with a hard switch to just CouchDB APIs.

Reply:
May be we deal with this separately as I see this distinction between CouchDB and Cloudant in few places like

1. CloudantRestClient
2. Vagrantfile
3. CouchDbRestClientTests

Reply:
This is vestigial at this point and can be removed in separate PR.